src/main/java/org/opentripplanner/transit/raptor/api/transit/RaptorTransfer.java

```diff
@@ -47,4 +47,13 @@
      */
     int durationInSeconds();
 
+    /**
+     * Does the access/transfer/egress contain transfers, which should increase the raptor round
+     * counter for the states?
+     *
+     * For accesses, determines at which round the access will be inserted to the RAPTOR state
+     */
+    default int extraRaptorRounds() {
+        return 0;
+    }
```

Comment:
I think what I would like to see as methods on this interface is:
```java
/** 
 * Is this {@link RaptorTransfer} is connected to the given {@code stop} by <b>transit</b>? For access and egress
 * paths we allow plugging in flex and other means of transport, witch might include one or more legs onboard a
 * vehicle. This method should return {@code true} if the leg connecting to the given stop is `onBoard` a public
 * transport or riding another kind of service like a taxi. This method apply to both access and egress, but is not
 * supported for public-transit-transfers. 
 */
default boolean connectedByPublicService() { return false; }

/** 
 * Return the number publicly available services for this {@link RaptorTransfer}. This is used by Raptor to get the 
 * number of transfers correct witch is part of the criteria used to keep optimal result. This method apply to both
 * access and egress, but is not supported for public-transit-transfers.    
 */
default int numberOfPublicServiceLegs() { return 0; } 

```

Reply:
At todays meeting we discussed if these had an effect on egress as well as access, and they do. At the destination there is a criteria for `numberOfTransfers` so we need to know about them for Flex egresses. I would like the Raptor API to be independent of the implementation, so the `round` is not something that the outside code should know about. That is why I changed the above method and instead pass in the information that Raptor can use to find the right place to insert the access.

Reply:
I use the term *Public Service* and *Public Transport*, because I would like to include taxi, city-bike and other forms of publicly available transport. We should probably discuss this with Andrew, but I think there is a bit of hassle for every service you use - so it become more fare to punish these services a little (adding extra transfers).